### PR TITLE
Analyzer speed improvements

### PR DESCRIPTION
**Pull request by zooba**
This PR is mostly about improving the analysis browser so it can export diffable versions of DBs.

The analyzer perf improvement comes from deferring all files from `*.test[s].*` packages into a second analysis. Effectively, we let test packages import from their surrounding package, but not feed analysis back into them. This results in significantly less results, most of which appear as fewer parameter types (which are basically unused anyway).

In a majority of cases, these look like improvements (losing incorrect information), while there are some cases where we lose correct information about some attribute types. Given the speed improvement in the initial analysis, this seems to be worthwhile.

We also now ensure that all analyzer packages are loaded when analyzing others, and allow the order to be overridden to ensure certain packages are analyzed first (addresses #1136).

### CODE REVIEW

File: Python/Product/AnalysisBrowser/FunctionView.cs [hunk 1]
```diff
@@ -37,29 +38,45 @@ public FunctionView(IModuleContext context, string name, IPythonFunction member,
...
+                }
 
-                            args[i].Add(parameters[i].Name);
+                var args = new List<HashSet<string>>();
+                var types = new List<HashSet<string>>();
+
+                foreach (var overload in _function.Overloads) {
+                    var parameters = overload.GetParameters();
+                    for (int i = 0; i < parameters.Length; ++i) {
+                        while (args.Count <= i) {
```
Comment (DinoV) [inline Python/Product/AnalysisBrowser/FunctionView.cs]: That's this while loop for?  Isn't this the same as args.Add(new HashSet<string>()); with no loop?

Comment (zooba) [inline Python/Product/AnalysisBrowser/FunctionView.cs]: Depends how many overloads we have, how many parameters each has, and what order we get to them in.

File: Python/Product/AnalysisBrowser/MultipleMemberView.cs [hunk 2]
```diff
@@ -44,5 +46,38 @@ class MultipleMemberView : MemberView {
...
+            TextWriter writer,
+            string currentIndent,
+            string indent,
+            Stack<IAnalysisItemView> exportStack,
+            out IEnumerable<IAnalysisItemView> exportChildren
+        ) {
+            var children = new HashSet<string>();
+            foreach (var c in Children) {
+                string displayValue = "";
+                if (c is ClassView) {
```
Comment (DinoV) [inline Python/Product/AnalysisBrowser/MultipleMemberView.cs]: Maybe have a DisplayValue virtual member?

**Comment by DinoV** [comment]
LGTM other than the slightly odd loop

File: Python/Product/AnalysisBrowser/MultipleMemberView.cs (hunk 2 above)
Comment (zooba) [inline Python/Product/AnalysisBrowser/MultipleMemberView.cs]: Considered that, but I'd still want the type filtering here. e.g. `ModuleRefView.DisplayValue` would have something valuable to display, but we'd still want to exclude it here for the purposes of the export.

**Comment by zooba** [comment]
The loop is basically unavoidable, and I don't really like any of the alternatives I can think of for a transpose operation like that.

**Pull request merged**